### PR TITLE
[fix](function) fix date_format and from_unixtime core when meet long format string

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

before:
crash

now:
```sql
mysql> SELECT  DATE_FORMAT('2011-06-24', '%f %V %f %l %V %I %S %p %w %r %j %f %l %I %D %w %j %D %e %s %V %f %D %M %s %X %U %v %c %u %x %r %j %a %h %s %m %a %v %u %b');
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| date_format(cast('2011-06-24' as DATETIMEV2(0)), '%f %V %f %l %V %I %S %p %w %r %j %f %l %I %D %w %j %D %e %s %V %f %D %M %s %X %U %v %c %u %x %r %j %a %h %s %m %a %v %u %b') |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| NULL                                                                                                                                                                           |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.80 sec)
```

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/runtime/vdatetime_value.h [hunk 1]
```diff
@@ -822,7 +823,8 @@ class DateV2Value {
         return val;
     }
 
-    bool to_format_string(const char* format, int len, char* to) const;
+    bool to_format_string(const char* format, int len, char* to,
+                          int max_valid_length = 10000) const;
```
Comment (zhiqiang-hhhh) [inline be/src/vec/runtime/vdatetime_value.h:827]: maybe no default value is better

**Review by zhiqiang-hhhh — APPROVED** [review]
LGTM

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/runtime/vdatetime_value.h (hunk 1 above)
Comment (zclllyybb) [reply be/src/vec/runtime/vdatetime_value.h]: good idea. fixed

File: be/src/olap/types.h [hunk 2]
```diff
@@ -1240,7 +1240,7 @@ struct FieldTypeTraits<FieldType::OLAP_FIELD_TYPE_DATEV2>
         string res;
         res.resize(12);
```
Comment (HappenLee) [inline be/src/olap/types.h:1241]: why here call reize and reserve? only need call one

File: be/src/vec/runtime/vdatetime_value.cpp [hunk 3]
```diff
@@ -3421,10 +3519,12 @@ void DateV2Value<T>::set_microsecond(uint32_t microsecond) {
 }
 
 template <typename T>
-bool DateV2Value<T>::to_format_string(const char* format, int len, char* to) const {
+bool DateV2Value<T>::to_format_string(const char* format, int len, char* to,
```
Comment (HappenLee) [inline be/src/vec/runtime/vdatetime_value.cpp:3522]: should simple the function, too much if check

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Mryange — APPROVED** [review]

**Pull request merged**